snuba/datasets/storages/events.py

```diff
@@ -0,0 +1,319 @@
+        ("geo_region", Nullable(String())),
+        ("geo_city", Nullable(String())),
+        ("sdk_name", Nullable(String())),
+        ("sdk_version", Nullable(String())),
+        ("type", Nullable(String())),
+        ("version", Nullable(String())),
+    ]
+    + metadata_columns
+    + promoted_context_columns
+    + promoted_tag_columns
+    + promoted_context_tag_columns
+    + [
+        # other tags
+        ("tags", Nested([("key", String()), ("value", String())])),
+        ("_tags_flattened", String()),
+        # other context
+        ("contexts", Nested([("key", String()), ("value", String())])),
+        # http interface
+        ("http_method", Nullable(String())),
+        ("http_referer", Nullable(String())),
+        # exception interface
+        (
+            "exception_stacks",
+            Nested(
+                [
+                    ("type", Nullable(String())),
+                    ("value", Nullable(String())),
+                    ("mechanism_type", Nullable(String())),
+                    ("mechanism_handled", Nullable(UInt(8))),
+                ]
+            ),
+        ),
+        (
+            "exception_frames",
+            Nested(
+                [
+                    ("abs_path", Nullable(String())),
+                    ("filename", Nullable(String())),
+                    ("package", Nullable(String())),
+                    ("module", Nullable(String())),
+                    ("function", Nullable(String())),
+                    ("in_app", Nullable(UInt(8))),
+                    ("colno", Nullable(UInt(32))),
+                    ("lineno", Nullable(UInt(32))),
+                    ("stack_level", UInt(16)),
+                ]
+            ),
+        ),
+        # These are columns we added later in the life of the (current) production
+        # database. They don't necessarily belong here in a logical/readability sense
+        # but they are here to match the order of columns in production becase
+        # `insert_distributed_sync` is very sensitive to column existence and ordering.
+        ("culprit", Nullable(String())),
+        ("sdk_integrations", Array(String())),
+        ("modules", Nested([("name", String()), ("version", String())])),
+    ]
+)
+
+sample_expr = "cityHash64(toString(event_id))"
+schema = ReplacingMergeTreeSchema(
+    columns=all_columns,
+    local_table_name="sentry_local",
+    dist_table_name="sentry_dist",
+    mandatory_conditions=[("deleted", "=", 0)],
+    prewhere_candidates=[
+        "event_id",
+        "group_id",
+        "tags[sentry:release]",
+        "message",
+        "environment",
+        "project_id",
+    ],
+    order_by="(project_id, toStartOfDay(timestamp), %s)" % sample_expr,
+    partition_by="(toMonday(timestamp), if(equals(retention_days, 30), 30, 90))",
+    version_column="deleted",
+    sample_expr=sample_expr,
+    migration_function=events_migrations,
+)
+
+
+def _get_promoted_columns() -> Mapping[str, FrozenSet[str]]:
```

Comment:
All these methods are moved from the events dataset. 
#774 removes all of these functions.